### PR TITLE
Curating Owners: pkg/cloudprovider

### PR DESCRIPTION
**Pull request by apelisse**
cc @runseb @justinsb @kerneltime @mikedanese @svanharmelen @anguslees @brendandburns @abrarshivani @imkin @luomiao @colemickens @ngtuna @dagnello @abithap

In an effort to expand the existing pool of reviewers and establish a
two-tiered review process (first someone lgtms and then someone
experienced in the project approves), we are adding new reviewers to
existing owners files.

If You Care About the Process:
------------------------------

We did this by algorithmically figuring out who’s contributed code to
the project and in what directories.  Unfortunately, that doesn’t work
well: people that have made mechanical code changes (e.g change the
copyright header across all directories) end up as reviewers in lots of
places.

Instead of using pure commit data, we generated an excessively large
list of reviewers and pruned based on all time commit data, recent
commit data and review data (number of PRs commented on).

At this point we have a decent list of reviewers, but it needs one last
pass for fine tuning.

Also, see https://github.com/kubernetes/contrib/issues/1389.

TLDR:
-----

As an owner of a sig/directory and a leader of the project, here’s what
we need from you:

1. Use PR https://github.com/kubernetes/kubernetes/pull/35715 as an example.

2. The pull-request is made editable, please edit the `OWNERS` file to
remove the names of people that shouldn't be reviewing code in the
future in the **reviewers** section. You probably do NOT need to modify
the **approvers** section. Names asre sorted by relevance, using some
secret statistics.

3. Notify me if you want some OWNERS file to be removed.  Being an
approver or reviewer of a parent directory makes you a reviewer/approver
of the subdirectories too, so not all OWNERS files may be necessary.

4. Please use ALIAS if you want to use the same list of people over and
over again (don't hesitate to ask me for help, or use the pull-request
above as an example)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36530)

### CODE REVIEW

**Comment by apelisse** [comment]
his name is missing from the list, but consider adding "ArtfulCoder"

**Comment by anguslees** [comment]
Lgtm, I presume I should accept lgtm from other parties involved before
"officially" lgtm-ing this.  So: indicate awareness/approval people!

**Comment by svanharmelen** [comment]
As one of the persons in the cc, I can verify I'm good and happy with this! So LGTM

**Comment by anguslees** [comment]
@apelisse: erm, so far that's a rather low response rate.  Want to break it up into smaller changes that we can approve/abandon separately? (or any other suggestions for moving forward?)

**Comment by apelisse** [comment]
Yeah, we can split this one. I'll gather current feedback and split

File: pkg/cloudprovider/OWNERS [hunk 1]
```diff
@@ -1,4 +1,43 @@
-assignees:
-  - mikedanese
-owners:
-  - mikedanese
+approvers:
+- mikedanese
+reviewers:
+- thockin
```
Comment (mikedanese) [inline pkg/cloudprovider/OWNERS]: These lists of reviewers look excessive, they include people who don't work often in pkg/cloudprovider or are no longer with the project

File: pkg/cloudprovider/providers/vsphere/OWNERS [hunk 2]
```diff
@@ -1,7 +1,7 @@
-maintainers:
-- dagnello
+approvers:
 - abithap
```
Comment (kerneltime) [inline pkg/cloudprovider/providers/vsphere/OWNERS]: Can we drop abithap and dagnello, from what I understand they are no longer working on this.
@abithap @dagnello

**Comment by apelisse** [comment]
@k8s-bot kops aws e2e test this

File: pkg/cloudprovider/providers/vsphere/OWNERS (hunk 2 above)
Comment (apelisse) [inline pkg/cloudprovider/providers/vsphere/OWNERS]: Thanks! I've done that and updated the pr.

**Comment by calebamiles** [comment]
@apelisse, how about limiting the scope of this PR to the format changes of the `OWNERS` files themselves. I don't know if it's worth attempting to address the concerns of @mikedanese in this PR given the talk of eventually moving `cloudprovider` out of tree. If anything creating empty alias for the various cloud platform focused SIGs could make sense if it was crucial to expand the pool of reviewers for `cloudprovider`. What do you think @mikedanese?

**Comment by calebamiles** [comment]
@apelisse, created https://github.com/apelisse/kubernetes/pull/1 which only updates the `OWNERS` file format and deletes the `OWNERS` file in `pkg/cloudprovider/providers` since it only contains `reviewers`. Assuming that PR merges are you fine with the change of file format @mikedanese and @kerneltime so we can call the migration effort complete?

**Pull request merged**